docs/source/integrations/iceberg_versioning.md

```diff
@@ -0,0 +1,229 @@
+    ):
+        self.table_name = table_name
+        self.namespace = namespace
+        self.catalog = load_catalog(catalog)
+        self.table_type = table_type
+        self.load_args = load_args
+        self.table = self._load_table(namespace, table_name)
+        self.save_args = save_args
+        self.scan_args = scan_args
+
+
+    def load(self):
+        if not self.table:
+            raise DatasetError("Table does not exist")
+        if self.scan_args:
+            scan = self.table.scan(**self.scan_args)
+        else:
+            scan = self.table.scan()
+        if self.table_type == "pandas":
+            return scan.to_pandas()
+        elif self.table_type == "arrow":
+            return scan.to_arrow()
+        elif self.table_type == "duckdb":
+            return scan.to_duckdb()
+        elif self.table_type == "daft":
+            return scan.to_daft()
+        else:
+            raise DatasetError("Table type not supported")
+
+    def _load_table(self, namespace, table_name):
+        try:
+            return self.catalog.load_table((namespace, table_name))
+        except NoSuchTableError:
+            return None
+
+    def save(self, data) -> None:
+        arrow = pa.Table.from_pandas(data)
+        if not self.table:
+            self.catalog.create_namespace_if_not_exists(self.namespace)
+            self.table = self.catalog.create_table((self.namespace, self.table_name), schema=arrow.schema)
+        if self.save_args.get("mode") == "overwrite":
+            self.table.overwrite(arrow)
+        elif self.save_args.get("mode") == "append":
+            self.table.append(arrow)
+        else:
+            raise DatasetError("Mode not supported")
+
+    def _describe(self) -> dict:
+        return {}
+
+    def exists(self):
+        return self.catalog.table_exists((self.namespace, self.table_name))
+
+    def inspect(self):
+        return self.table.inspect
+```
+
+This dataset allows you to load Iceberg tables as `pandas`, `arrow`, `duckdb`, or `daft` objects. You can also load a subset of the table by passing the `scan_args` parameter to the dataset. The `save()` method allows you to save a dataframe as an Iceberg table. You can specify the mode of saving the table by passing the `save_args` parameter to the dataset. The `inspect()` method returns an `InspectTable` object which contains metadata about the table.
+You can also update the code to extend the functionality of the dataset to support more features of Iceberg tables. Refer to the [Iceberg API documentation](https://py.iceberg.apache.org/api/) to see what you can do with the `Table` object.
+
+## Using Iceberg tables in the catalog
+## Using Iceberg tables in the catalog
+
+### Save the dataset as an Iceberg table
+
+Now update your catalog in `conf/base/catalog.yml` to update the `model_input_table` dataset to use the custom `PyIcebergDataset`:
+
+```yaml
+model_input_table:
+  type: kedro_iceberg.pyiceberg_dataset.PyIcebergDataset
```

Comment:
Where is this `kedro_iceberg` project?

Reply:
I've updated the project creation bit on top to include the project name